### PR TITLE
New reports : Stock Balance Multi-UOM

### PR DESCRIPTION
**Pull request by lelsheri**
Report : Stock Balance Multi-UOM :
========================
to let customer view the Stock balances with default UOM + Purchase UOM + Sales UOM
as my client request.

Please read the pull request checklist to make sure your changes are merged: https://github.com/frappe/erpnext/wiki/Pull-Request-Checklist

### CODE REVIEW

**Comment by lelsheri** [comment]
Any feedback.

File: erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+	if filters.get('show_variant_attributes', 0) == 1:
+		columns += ["{}:Data:100".format(i) for i in get_variants_attributes()]
+
+	return columns, data
+
+def get_columns():
+	"""return columns"""
+
+	columns = [
```
Comment (rmehta) [inline erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py]: please use the new API with JSON structures

File: erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+	conditions = get_conditions(filters)
+
+	return frappe.db.sql("""
+		select
+			sle.item_code, warehouse, sle.posting_date, sle.actual_qty, sle.valuation_rate,
+			sle.company, sle.voucher_type, sle.qty_after_transaction, sle.stock_value_difference,
+			(SELECT uomc.conversion_factor
+             FROM `tabUOM Conversion Detail` uomc,`tabItem` itm
+             WHERE uomc.parent = sle.item_code
+                AND itm.name = sle.item_code
```
Comment (rmehta) [inline erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py]: inconsistent indenting

**Comment by rmehta** [comment]
@lelsheri 👍 for first contribution. How do we know if this works? Can you add a test case?

Also better to call the report "Stock Balance in Multiple UOMs"

**Review by rmehta — CHANGES_REQUESTED** [review]

File: erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+				qty_dict.out_qty += abs(qty_diff)
+				qty_dict.out_val += abs(value_diff)
+
+		qty_dict.val_rate = d.valuation_rate
+		qty_dict.bal_qty += qty_diff
+		qty_dict.bal_val += value_diff
+
+	if d.purchase_uom_conversion_factor is None:
+		   qty_dict.bal_qty_p = 0
+		elif d.purchase_uom_conversion_factor != 0:
```
Comment (Mangesh-Khairnar) [inline erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py]: fix the indentations to match with the outer if conditions.

File: erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+
+def get_variants_attributes():
+	'''Return all item variant attributes.'''
+	return [i.name for i in frappe.get_all('Item Attribute')]
+
+def get_variant_values_for(items):
+	'''Returns variant values for items.'''
+	attribute_map = {}
+	for attr in frappe.db.sql('''select parent, attribute, attribute_value
+		from `tabItem Variant Attribute` where parent in (%s)
```
Comment (Mangesh-Khairnar) [inline erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py]: Please follow the indentation guidelines:

https://github.com/frappe/erpnext/wiki/Coding-Standards#use-simple-indenting

File: erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+		   qty_dict.bal_qty_p = 0
+		elif d.purchase_uom_conversion_factor != 0:
+			qty_dict.bal_qty_p = qty_dict.bal_qty / d.purchase_uom_conversion_factor
+		else: qty_dict.bal_qty_p = 0
+
+	if d.sales_uom_conversion_factor is None:
+		   qty_dict.bal_qty_s = 0
+		elif d.sales_uom_conversion_factor != 0:
+		   qty_dict.bal_qty_s = qty_dict.bal_qty / d.sales_uom_conversion_factor
+		else: qty_dict.bal_qty_s = 0
```
Comment (Mangesh-Khairnar) [inline erpnext/stock/report/stock_balance_multi_uom/stock_balance_multi_uom.py]: indentation doesn't match. Incorrect logic.

**Review by Mangesh-Khairnar — CHANGES_REQUESTED** [review]

**Comment by sagarvora** [comment]
Closing. Please reopen after making necessary changes.

**Pull request closed without merging**